data-generation/src/main/scala/UpdateDistributions.scala

```diff
@@ -0,0 +1,183 @@
+import DistributionType.DistributionType
+import UpdateDistributionMode.UpdateDistributionMode
+import breeze.plot.{Figure, hist}
+import breeze.stats.distributions.{Gaussian, LogNormal}
+import org.apache.spark.SparkContext
+import org.apache.spark.graphx.{Edge, Graph, VertexId}
+
+import scala.reflect.ClassTag
+
+object UpdateDistributionMode extends Enumeration {
+  type UpdateDistributionMode = Value
+  val Uniform, PositiveCorrelation, NegativeCorrelation = Value
+}
+
+object DistributionType extends Enumeration {
+  type DistributionType = Value
+  val LogNormal, Gaussian = Value
+}
+
+object UpdateDistributions {
+
+  /** Add an update count to all nodes
+   *
+   * @param sc           A SparkContext
+   * @param graph        A graph
+   * @param mode         Mode for distributing the weights based on vertex degree
+   * @param distribution Some probability distribution
+   * @param mu           Expected value
+   * @param sigma        Standard deviation
+   * @return */
+  def addVertexUpdateDistribution[VD, ED: ClassTag](sc: SparkContext, graph: Graph[VD, ED], mode: UpdateDistributionMode, distribution: DistributionType, mu: Int, sigma: Double): Graph[Int, ED] = {
+    val graphWithDegree = graph
+      .vertices
+      .zip(graph.ops.degrees)
+      .map(vertex => (vertex._1._1, vertex._2._2))
```

Comment:
Scala er et helt utrolig stygt språk lærer jeg ettterhvert. Mer idiomatisk så kunne dette vært. Det du har er tydeligere
```suggestion
      .map((_._1._1, _._2._2))
```